tokio-util/src/udp/mod.rs

```diff
@@ -1,4 +1,62 @@
 //! UDP framing
 
+use std::{
+    borrow::Borrow,
+    io,
+    net::SocketAddr,
+    task::{Context, Poll},
+};
+use tokio::{io::ReadBuf, net::UdpSocket};
+
 mod frame;
 pub use frame::UdpFramed;
+
+/// Types that support receiving datagrams.
+///
+/// This trait is implemented for any types that implement [`Borrow`]<[`UdpSocket`]>.
+pub trait PollRecvFrom {
+    /// Attempts to receive a single datagram on the socket.
+    ///
+    /// See [`UdpSocket::poll_recv_from()`] for more information.
+    fn poll_recv_from(
+        &mut self,
+        cx: &mut Context<'_>,
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<io::Result<SocketAddr>>;
+}
+
+impl<T: Borrow<UdpSocket>> PollRecvFrom for T {
+    fn poll_recv_from(
+        &mut self,
+        cx: &mut Context<'_>,
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<io::Result<SocketAddr>> {
+        (*self).borrow().poll_recv_from(cx, buf)
+    }
+}
```

Comment:
Instead of doing this, maybe this should just be added to the list for the next breaking tokio-util release.

Reply:
There’s also an option to do both: merge this now and remove the blanket impl on the next breaking release.